### PR TITLE
feat: add 3 tier app, usage of userdata, sec grps

### PR DESCRIPTION
**Pull request by esn89**
Adds another example showing:

1. loading user data into autoscaling groups
2. create an application load balancer and the listeners/target groups
3. dockerized application and how to start them
4. other security groups as a source of ingress to other security groups.

Fixes # 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by rhboyd** [comment]
Can you rename the folder to something a bit more descriptive? It’s understood to be a CDK app since it’s in the CDK examples repository.

**Comment by esn89** [comment]
@rhboyd I sure can.  Thank you for your reply.

**Comment by esn89** [comment]
@rhboyd 

How does it look now?

Could you also please help me out with the travis build script?  I imagine that it is trying to build out my code on a test AWS environment.  

I am doing `cdk deploy $(cdk ls)` on my local machine and that seems to be okay.  Is there anything else I need to take into account of, to make sure the build script passes?   Thanks!

File: python/docker-app-with-asg-alb/dockerized_app_cdk/nextcloud_docker_cdk_stack.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from aws_cdk import core
```
[the remark this answers is not in the document]
Comment (esn89) [reply python/docker-app-with-asg-alb/dockerized_app_cdk/nextcloud_docker_cdk_stack.py]: Good catch, this was a test file I made.  I will delete it.

File: python/docker-app-with-asg-alb/setup.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import setuptools
+
+
+with open("README.md") as fp:
+    long_description = fp.read()
+
+
+setuptools.setup(
+    name="dockerized_app_cdk",
+    version="0.0.1",
```
[the remark this answers is not in the document]
Comment (esn89) [reply python/docker-app-with-asg-alb/setup.py]: Done!

File: python/docker-app-with-asg-alb/setup.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    description="An empty CDK Python app",
```
[the remark this answers is not in the document]
Comment (esn89) [reply python/docker-app-with-asg-alb/setup.py]: Done!

File: python/docker-app-with-asg-alb/setup.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+
+    author="author",
```
[the remark this answers is not in the document]
Comment (esn89) [reply python/docker-app-with-asg-alb/setup.py]: Will do, thank you.  Also done :)

**Comment by esn89** [comment]
@NGL321 

Thanks for the kind words.  I have done the revisions, hope it is all good now :)

**Comment by esn89** [comment]
@NGL321 Thanks!  Can we get a merge? :)

**Pull request merged**